default/scripting/ship_hulls/asteroid/SH_CRYSTALLIZED_ASTEROID.focs.txt

```diff
@@ -35,6 +35,20 @@ Hull
                 OwnedBy empire = Source.Owner
             ]
         ]
+        ResourceSupplyConnected empire = Source.Owner condition = And [
+             Building name = "BLD_SHIPYARD_AST"
+             Or [
+                 OwnedBy empire = Source.Owner
+                 OwnedBy affiliation = AllyOf empire = Source.Owner
+             ]
+        ]
+        ResourceSupplyConnected empire = Source.Owner condition = And [
+             Building name = "BLD_SHIPYARD_AST_REF"
+             Or [
+                 OwnedBy empire = Source.Owner
+                 OwnedBy affiliation = AllyOf empire = Source.Owner
+             ]
```

Comment:
This is slightly different as it's the reformation processor but it's basically the same problem, it already requires it be in system.